### ISSUE TITLE
useSearchField keyboard accessibility error on clear button

### ISSUE THREAD

**Issue by CarynPagelTR** [adobe/react-spectrum#3231]
# 🐛 Bug Report

The clear search button on [useSearchField](https://react-spectrum.adobe.com/react-aria/useSearchField.html) is not keyboard accessible. A few issues:
* Keyboard users can only clear the field manually or by pressing Esc. There is no way for a keyboard user to know Esc key will clear the field. The Esc key is generally reserved for collapsing or closing content, not for clearing text.
* Screen reader users have the clear button in their list of form controls on the page, but they cannot use it because it will not receive focus.

## 🤔 Expected Behavior

The clear button can receive keyboard focus.

## 😯 Current Behavior

The clear button cannot receive keyboard focus.

## 💁 Possible Solution

Remove `tabindex="-1"` from the clear button. Allow the clear button to receive keyboard focus.

## 🔦 Context

This issue is causing an accessibility issue on our website. We have to wait for it to be fixed in the library, or possibly override it on our site.

## 💻 Code Sample

Current HTML: `<button tabindex="-1" ... >`
Suggested HTML: `<button ... >`

## 🌍 Your Environment

not applicable

## 🧢 Your Company/Team

Thomson Reuters

## 🕷 Tracking Issue (optional)

not applicable

**Comment by devongovett** [comment]
I don't think this would be expected. We follow native browser and platform behavior here. For example, [`<input type="search">`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search) often includes a clear button provided by the browser or operating system, which does not receive focus, and the Escape key can be used to clear the field. This is a very common user experience, so users may actually expect this behavior as they have come to learn it across websites and applications on their system. In addition, if keyboard users are unaware of Escape, they can always backspace, or select all and then backspace to delete the text in the field.

Making the clear button tabbable also has the downside of cluttering the tab order, adding an additional tab stop that users must get past before getting to the next field. For these reasons, making the clear button tabbable is not usually expected or recommended. See https://scottaohara.github.io/clear-text-field-button/ for another example and explanation.

**Comment by CarynPagelTR** [comment]
Has the Adobe team tested this component with keyboard and screen reader users? What about the fact that the button is listed in screen reader form controls, but doesn't actually work for them? 

I talked to my team today, and gave them the info you've shared here. We're not comfortable having an interactive element on the page that's not focusable. If you will not update the component, we will be overriding on our end to remove `tabindex="-1"`.

**Comment by devongovett** [comment]
Going to close this. If you have more details, feel free to share. Otherwise, if you have a good reason to make the button tabbable, you can always override the `tabIndex` returned by `useSearchField`.

**Comment by CarynPagelTR** [comment]
> Can you share more about the issue with form controls you're seeing? The button is focusable, just not tabbable. Using screen reader navigation, it is still possible to focus the the clear button. For example, with VoiceOver you can use ctrl + option + right arrow to move the VO cursor to the button.

Yes you're right the button can be activated via form controls in screen reader. However, screen reader users that are accustomed to tabbing through all interactive elements may not know the clear button exists. Some feedback from my team member who is a native screen reader user: "It's an issue because it's on screen and you cannot reach it and alternative is not obvious. I expected escape to close the search bar, not clear text. Why would these be exceptions?" 

Also, sighted keyboard users can see the button, but there is no way for them to activate it.

> Not everything should be tabbable, otherwise it becomes difficult to navigate around the page because there are so many elements in the tab order.

Yes there are many cases where elements are purposely not in the tab order. Example: HTML natively does this for radio buttons - only the first (or selected) radio button receives focus using Tab and the other radio buttons can be accessed via arrows. In the case of the clear button, there is no alternative way to focus on it, using arrows or any other keyboard method.
